### PR TITLE
YLP-839 Missing error string for translation

### PR DESCRIPTION
**Pull request by yvon-dblg**

### CODE REVIEW

**Review by franck-fourel — COMMENTED** [review]
looks like the scope of this PR is above the original scope of the JIRA issue
anyway it looks good
couple of comments on some piece of code

File: packages/yourloops/lib/auth/api.ts [hunk 1]
```diff
@@ -60,78 +61,78 @@ function format(user: any): User {
  * @return {Promise<User>} Return the logged-in user or a promise rejection.
  */
 async function authenticate(username: string, password: string, traceToken: string): Promise<Session> {
-  let reason: string | null = null;
-
   if (!_.isString(username) || _.isEmpty(username)) {
-    reason = t("no-username") as string;
-    return Promise.reject(new Error(reason));
+    sendMetrics("Login", "no-username");
+    return Promise.reject(new Error("no-username"));
```
[the remark this answers is not in the document]
Comment (yvon-dblg) [reply packages/yourloops/lib/auth/api.ts:66]: Good question. I don't have the answer.

File: packages/yourloops/lib/auth/api.ts [hunk 2]
```diff
@@ -143,52 +144,56 @@ async function authenticate(username: string, password: string, traceToken: stri
  * @return {Promise<User>} Return the logged-in user or a promise rejection.
  */
 async function signup(username: string, password: string, role: UserRoles, traceToken: string): Promise<Session> {
-  let reason: string | null = null;
-
   if (!_.isString(username) || _.isEmpty(username)) {
-    reason = t("no-username") as string;
-    return Promise.reject(new Error(reason));
+    return Promise.reject(new Error("no-username"));
```
[the remark this answers is not in the document]
Comment (yvon-dblg) [reply packages/yourloops/lib/auth/api.ts:148]: In the UI the check for the login & password not empty is done before calling this method.
It's more a last resort check, which should never pass the dev stage.
You may call it a good practice: check the parameters of a function to detect early some bugs.

**Review by franck-fourel — APPROVED** [review]
LGTM

**Pull request merged**